src/api/ppx_deriving.cppo.ml

```diff
@@ -142,6 +143,44 @@ let create =
 let string_of_core_type typ =
   Format.asprintf "%a" Pprintast.core_type { typ with ptyp_attributes = [] }
 
+type constant =
+  #if OCAML_VERSION >= (4, 03, 0)
+    Parsetree.constant
+  #else
+    Asttypes.constant
+  #endif
+
+let string_of_constant_opt (constant : constant) : string option =
+  match constant with
+  #if OCAML_VERSION >= (4, 11, 0)
+  | Pconst_string (s, _, _) ->
+  #else
+  | Pconst_string (s, _) ->
+  #endif
```

Comment:
Nitpick: it would be a bit more idiomatic in this file to define `Pconst_string` as a macro (or `Pconst_string_patt` if we have reasons to believe that we would also need it as a term-former) to have the same interface on all versions, at the top of the file (where we have other definitions in this style), and use the macro here.

```ocaml
#if OCAML_VERSION >= (4, 11, 0)
#define Pconst_string(s, loc) Pconst_string(s, loc, _)
#endif
```